### PR TITLE
Rerun analysis

### PR DESCRIPTION
**Pull request by MikeLippincott**
This PR reruns all analysis after fixing a Cytotable and Pycytominer error!

### CODE REVIEW

File: 1.Exploratory_Data_Analysis/all_anova_call_local.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+conda activate Interstellar_python
+
+jupyter nbconvert --to=script --FilesWriter.build_directory=scripts/ notebooks/*.ipynb
+
+CELL_TYPE="SHSY5Y"
```
Comment (gwaybio) [inline 1.Exploratory_Data_Analysis/all_anova_call_local.sh:7]: Do you also run this with PBMCs? You might also consider looping over them like you do in line 18.

File: 1.Exploratory_Data_Analysis/features/PBMC_feature_index.txt [hunk 2]
```diff
@@ -50,16 +48,15 @@ Cytoplasm_Correlation_Overlap_CorrDNA_CorrPM
 Cytoplasm_Correlation_Overlap_CorrGasdermin_CorrMito
 Cytoplasm_Correlation_Overlap_CorrGasdermin_CorrPM
 Cytoplasm_Correlation_Overlap_CorrMito_CorrPM
-Cytoplasm_Correlation_RWC_CorrDNA_CorrER
-Cytoplasm_Correlation_RWC_CorrDNA_CorrMito
-Cytoplasm_Correlation_RWC_CorrDNA_CorrPM
+Cytoplasm_Correlation_RWC_CorrER_CorrDNA
```
Comment (gwaybio) [inline 1.Exploratory_Data_Analysis/features/PBMC_feature_index.txt:51]: Do you know why these are different? Is it b/c feature selection is different? (just curious)

File: 1.Exploratory_Data_Analysis/notebooks/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Exploratory data analysis
+
+This folder contains notebooks that demonstrate how to perform exploratory data analysis (EDA) on a dataset.
+EDA is an essential step in the data analysis process.
+It allows you to summarize the main characteristics of the data, often with visual methods. This helps you understand the data and the relationships between the variables.
```
Comment (gwaybio) [inline 1.Exploratory_Data_Analysis/notebooks/README.md:5]:
```suggestion
It allows you to summarize the main characteristics of the data, often with visual methods.
This helps you understand the data and the relationships between the variables.
```

File: 1.Exploratory_Data_Analysis/notebooks/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+Due to the exploratory nature of the analysis, some analysis are secondary and were not included in the manuscript that this repository is for.
+However, they are still useful for understanding the data and the relationships between the variables.
+
+The main analysis that are needed for the manuscript are:
+
+1. [UMAP](1.umap_analysis_plate2.ipynb)
```
Comment (gwaybio) [inline 1.Exploratory_Data_Analysis/notebooks/README.md:12]: consider a brief blurb describing what "plate 2" data are. Alternatively, you can avoid saying "plate 2" here and just state directly what plate 2 is.

File: 1.Exploratory_Data_Analysis/notebooks/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    This notebook extracts the UMAP embeddings of the plate 2 data and visualizes the embeddings.
+2. [Cell Counts](4.cell_count_analysis.ipynb)
+    This notebook extracts the cell counts of the plate 2 data and visualizes the cell counts.
+3. [Feature index](8.0_create_feature_index.ipynb)
+    This notebook creates a feature index for the plate 2 data.
+4. [ANOVA](8.1_anova_all_groupings.ipynb)
+    This notebook performs ANOVA on the plate 2 data.
+5. [Combine ANOVA](8.2_combine_files.ipynb)
+    This notebook combines the ANOVA results from the previous notebook.
+6. [Subset ANOVA](9.subset_umap.ipynb)
```
Comment (gwaybio) [inline 1.Exploratory_Data_Analysis/notebooks/README.md:22]: Subsets the UMAP data how and why?

File: 1.Exploratory_Data_Analysis/scripts/1.umap_analysis_plate2.py [hunk 6]
```diff
@@ -3,7 +3,7 @@
 
 # ## Plate 2
 
-# In[1]:
+# In[ ]:
```
Comment (gwaybio) [inline 1.Exploratory_Data_Analysis/scripts/1.umap_analysis_plate2.py:6]: This notebook appears to not be sequentially executed

File: 1.Exploratory_Data_Analysis/scripts/4.cell_count_analysis.py [hunk 7]
```diff
@@ -25,14 +25,14 @@
...
-path = pathlib.Path(f"../data/{celltype}_preprocessed_sc_norm.parquet")
+path = pathlib.Path(f"../../data/{celltype}_preprocessed_sc_norm.parquet")
 
 
 # In[4]:
 
 
 # import data
-df = pq.read_table(path).to_pandas()
+df = pd.read_parquet(path)
```
Comment (gwaybio) [inline 1.Exploratory_Data_Analysis/scripts/4.cell_count_analysis.py:35]: probably ok for EDA and this project, but improving variable naming (to describe what the dataframe holds) will help with clarity in your next project

File: 1.Exploratory_Data_Analysis/scripts/8.0_create_feature_index.py [hunk 8]
```diff
@@ -27,12 +26,18 @@
...
 
-# parse command line arguments
-parser = argparse.ArgumentParser()
-parser.add_argument("--cell_type", help="cell type to analyze", type=str, default="all")
+cell_type = "SHSY5Y"
 
-args = parser.parse_args()
-cell_type = args.cell_type
+
+# In[ ]:
```
Comment (gwaybio) [inline 1.Exploratory_Data_Analysis/scripts/8.0_create_feature_index.py:32]: this file doesn't appear to be sequentially executed either

File: 1.Exploratory_Data_Analysis/scripts/9.subset_umap.py [hunk 9]
```diff
@@ -63,7 +63,13 @@
...
 
 
-# In[7]:
+# Where
+# a_h = apoptosis vs healthy
+# a_p = apoptosis vs pyroptosis
+# h_p = healthy vs pyroptosis
+#
+
+# In[ ]:
```
Comment (gwaybio) [inline 1.Exploratory_Data_Analysis/scripts/9.subset_umap.py:72]: Please confirm sequential execution on all notebooks - I will stop making this comment here on!

File: 4.sc_Morphology_Neural_Network_MLP_Model/1.model_training/scripts/Multi_Classifier_Call.sh [hunk 10]
```diff
@@ -14,9 +14,9 @@ module load anaconda
 
 conda activate Interstellar_python
 
-cell_types=( SHSY5Y PBMC )
+cell_types=( PBMC )
 model_names=( MultiClass_MLP )
-shuffles=( True False )
+shuffles=( True )
```
Comment (gwaybio) [inline 4.sc_Morphology_Neural_Network_MLP_Model/1.model_training/scripts/Multi_Classifier_Call.sh:19]: This probably should not have changed, correct? In terms of reproducibility

File: 6.bulk_Morphology_Elastic_Network/0.split_data/scripts/split_data_call.sh [hunk 11]
```diff
@@ -8,14 +8,14 @@
...
-#SBATCH --array=1-2
+#SBATCH --array=1
 
 module load anaconda
 
 conda activate Interstellar
 
 # define the search parameters
-cell_types=( "SHSY5Y" "PBMC" )
+cell_types=( "PBMC" )
```
Comment (gwaybio) [inline 6.bulk_Morphology_Elastic_Network/0.split_data/scripts/split_data_call.sh:18]: I think this also probably shouldn't change.

File: 6.bulk_Morphology_Elastic_Network/2.test_model/notebooks/2.combine_regression_tests.ipynb [hunk 12]
```diff
@@ -14,9 +14,35 @@
...
+            "execution_count": 2,
             "metadata": {},
-            "outputs": [],
+            "outputs": [
+                {
+                    "name": "stderr",
+                    "output_type": "stream",
+                    "text": [
+                        "usage: ipykernel_launcher.py [-h] [--cell_type CELL_TYPE]\n",
+                        "ipykernel_launcher.py: error: unrecognized arguments: --f=/home/lippincm/.local/share/jupyter/runtime/kernel-v2-3427334WMucDBDEj4MH.json\n"
```
Comment (gwaybio) [inline 6.bulk_Morphology_Elastic_Network/2.test_model/notebooks/2.combine_regression_tests.ipynb:25]: flagging error here

File: 6.bulk_Morphology_Elastic_Network/2.test_model/scripts/1.test_regression_multi_output.py [hunk 13]
```diff
@@ -227,67 +210,65 @@
...
-    df["log10_explained_variance"] = -np.log10(df["explained_variance"])
-
-    # replace "[NSU]" with """
-    df["cytokine"] = df["cytokine"].replace("[ \[\]NSU]", "", regex=True)
-    df["cytokine"] = df["cytokine"].replace(" ", "_", regex=True)
-
-    # concat the dataframes
-    results_df = pd.concat([results_df, df], axis=0)
+data_x = test_data_x
+data_y = test_data_y
```
Comment (gwaybio) [inline 6.bulk_Morphology_Elastic_Network/2.test_model/scripts/1.test_regression_multi_output.py:214]: this is a bit dangerous (without a deep copy) if you use any of the test data later in this notebook

File: 6.bulk_Morphology_Elastic_Network/2.test_model/scripts/test_model_call_cluster.sh [hunk 14]
```diff
@@ -19,7 +19,7 @@ readarray -t cytokine_array < $filename
 
 
 shuffles=( "True" "False" )
-cell_types=( SHSY5Y PBMC )
+cell_types=( PBMC )
```
Comment (gwaybio) [inline 6.bulk_Morphology_Elastic_Network/2.test_model/scripts/test_model_call_cluster.sh:22]: Again, you might not want these to change... I'm not going to make these comments from here on out - please confirm next time you go through!

File: 6.bulk_Morphology_Elastic_Network/2.test_model/scripts/test_model_call_local.sh [hunk 15]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# This script is used to train the regression models for the elastic network
+
+ jupyter nbconvert --to python --output-dir=. ../notebooks/*.ipynb
+
+conda activate Interstellar_python
+
+# get the array of cytokiens
```
Comment (gwaybio) [inline 6.bulk_Morphology_Elastic_Network/2.test_model/scripts/test_model_call_local.sh:8]:
```suggestion
# get the array of cytokines
```

**Review by gwaybio — APPROVED** [review]
LGTM, i made three categories of suggestions

1. Making sure to sequentially execute notebooks (or maybe the solution is to rerun nbconvert)
2. confirming some changes are intentional. Some files appear to be changed for convenience (e.g., only running in PBMCs). This is definitely the way to go to speed progress, but we want to minimize costs to reproducibility.
3. Minor clarification points

File: 1.Exploratory_Data_Analysis/features/PBMC_feature_index.txt (hunk 2 above)
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/features/PBMC_feature_index.txt:51]: I believe so, yes.

File: 1.Exploratory_Data_Analysis/notebooks/README.md (hunk 4 above)
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/notebooks/README.md:13]: Adding link to the repo!

File: 6.bulk_Morphology_Elastic_Network/2.test_model/notebooks/2.combine_regression_tests.ipynb (hunk 12 above)
Comment (MikeLippincott) [reply 6.bulk_Morphology_Elastic_Network/2.test_model/notebooks/2.combine_regression_tests.ipynb:25]: Removing error here.

File: 6.bulk_Morphology_Elastic_Network/2.test_model/scripts/1.test_regression_multi_output.py (hunk 13 above)
Comment (MikeLippincott) [reply 6.bulk_Morphology_Elastic_Network/2.test_model/scripts/1.test_regression_multi_output.py:214]: I learned what a deep copy is. I will implement in the future

**Pull request merged**